processor/internal/processor.go

```diff
@@ -14,4 +17,7 @@ type Settings struct {
 
 	// BuildInfo can be used by components for informational purposes
 	BuildInfo component.BuildInfo
+
+	// PipelineID indicates which pipeline contains this processor.
+	PipelineID pipeline.ID
```

Comment:
Settings are passed to the factory method correct? Which already signals which "signal" is this for. I am confused why adding it here?

Reply:
This indicates the pipeline ID, not the signal. The rationale for adding pipeline ID is explained in https://github.com/open-telemetry/opentelemetry-collector/issues/10908